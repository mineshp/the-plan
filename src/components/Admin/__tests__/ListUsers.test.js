import React from 'react';
import renderer from 'react-test-renderer';
import ListUsersComponent from '../ListUsers';
import { mockListUsers } from '../../../helpers/test/testData/controlCenterData';

const mockHandleDeleteUser = jest.fn();
const mockHandleResetPassword = jest.fn();

describe('Admin > Manage > ListUsers', () => {
    const mockUsers = mockListUsers();
    it('renders ListUsers component', () => {
        const tree = renderer.create(
            <ListUsersComponent
                users={mockUsers}
                handleDeleteUser={mockHandleDeleteUser}
                handleResetPwd={mockHandleResetPassword}
            />
        );
        expect(tree).toMatchSnapshot();
    });
});
