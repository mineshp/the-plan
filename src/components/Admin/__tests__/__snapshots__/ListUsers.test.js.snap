// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin > Manage > ListUsers renders ListUsers component 1`] = `
<div
  className="ui container"
>
  <table
    className="ui celled striped table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          Username
        </th>
        <th
          className=""
        >
          Email
        </th>
        <th
          className=""
        >
          Profiles Assigned
        </th>
        <th
          className=""
        >
          Password
        </th>
        <th
          className=""
        >
          Delete
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className="warning"
      >
        <td
          className="collapsing"
        >
          <i
            aria-hidden="true"
            className="graduation icon"
          />
           
          testUser
        </td>
        <td
          className=""
        >
          test@mytestemail.com
        </td>
        <td
          className=""
        >
          <a
            className="ui pink label"
            onClick={[Function]}
          >
             
            TESTA
          </a>
          <a
            className="ui pink label"
            onClick={[Function]}
          >
             
            TESTB
          </a>
        </td>
        <td
          className="collapsing"
        >
          <button
            className="ui green small button"
            disabled={undefined}
            id="1234567"
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
          >
            Reset Password
          </button>
        </td>
        <td
          className="collapsing"
        >
          <button
            className="ui pink small disabled button"
            disabled={true}
            id="1234567"
            onClick={[Function]}
            role="button"
            tabIndex={-1}
          >
            Delete User
          </button>
        </td>
      </tr>
      <tr
        className=""
      >
        <td
          className="collapsing"
        >
          <i
            aria-hidden="true"
            className="user icon"
          />
           
          god
        </td>
        <td
          className=""
        >
          test@anotheremail.com
        </td>
        <td
          className=""
        />
        <td
          className="collapsing"
        >
          <button
            className="ui green small button"
            disabled={undefined}
            id="1234568"
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
          >
            Reset Password
          </button>
        </td>
        <td
          className="collapsing"
        >
          <button
            className="ui pink small button"
            disabled={undefined}
            id="1234568"
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
          >
            Delete User
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
